<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Reproductor de Música</string>
  </property>
  <property name="styleSheet">
   <string>
    /* Estilo general de la ventana */
    QMainWindow {
        background-color: #1E1E1E;
        color: #FFFFFF;
    }

    /* Lista de reproducción */
    QTreeWidget {
        background-color: #2D2D2D;
        border: none;
        color: #FFFFFF;
    }
    QTreeWidget::item {
        height: 25px;
        padding: 4px;
    }
    QTreeWidget::item:selected {
        background-color: #0078D4;
    }
    QTreeWidget::item:hover {
        background-color: #3D3D3D;
    }
    QHeaderView::section {
        background-color: #2D2D2D;
        color: #FFFFFF;
        padding: 4px;
        border: none;
    }

    /* Controles de reproducción */
    QPushButton {
        background-color: transparent;
        color: #FFFFFF;
        border: none;
        font-size: 16px;
        padding: 8px;
        min-width: 32px;
    }
    QPushButton:hover {
        background-color: #3D3D3D;
        border-radius: 4px;
    }
    QPushButton:pressed {
        background-color: #0078D4;
    }

    /* Sliders */
    QSlider::groove:horizontal {
        border: none;
        height: 4px;
        background: #3D3D3D;
        margin: 2px 0;
    }
    QSlider::handle:horizontal {
        background: #0078D4;
        width: 12px;
        height: 12px;
        margin: -4px 0;
        border-radius: 6px;
    }
    QSlider::sub-page:horizontal {
        background: #0078D4;
    }
    QSlider::handle:horizontal:hover {
        background: #2B9FFF;
    }

    /* Etiquetas de tiempo */
    QLabel {
        color: #FFFFFF;
    }

    /* ComboBox de velocidad */
    QComboBox {
        background-color: #3D3D3D;
        color: #FFFFFF;
        border: none;
        padding: 4px;
        border-radius: 4px;
    }
    QComboBox::drop-down {
        border: none;
        width: 20px;
    }
    QComboBox::down-arrow {
        image: url(:/icons/down-arrow.png);
    }
    QComboBox:hover {
        background-color: #4D4D4D;
    }
    QComboBox QAbstractItemView {
        background-color: #2D2D2D;
        color: #FFFFFF;
        selection-background-color: #0078D4;
        selection-color: #FFFFFF;
        border: none;
    }

    /* Barra de estado */
    QStatusBar {
        background-color: #1E1E1E;
        color: #FFFFFF;
    }
    QStatusBar::item {
        border: none;
    }

    /* Scrollbars */
    QScrollBar:vertical {
        background: #2D2D2D;
        width: 10px;
        margin: 0;
    }
    QScrollBar::handle:vertical {
        background: #3D3D3D;
        min-height: 20px;
        border-radius: 5px;
    }
    QScrollBar::add-line:vertical,
    QScrollBar::sub-line:vertical {
        height: 0;
    }
    QScrollBar::up-arrow:vertical,
    QScrollBar::down-arrow:vertical {
        height: 0;
    }
    QScrollBar::add-page:vertical,
    QScrollBar::sub-page:vertical {
        background: none;
    }
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTreeWidget" name="playlistWidget">
      <property name="acceptDrops">
       <bool>true</bool>
      </property>
      <property name="dragDropMode">
       <enum>QAbstractItemView::DropOnly</enum>
      </property>
      <property name="defaultDropAction">
       <enum>Qt::CopyAction</enum>
      </property>
      <column>
       <property name="text">
        <string>Título</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Artista</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Álbum</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Duración</string>
       </property>
      </column>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="progressLayout">
      <item>
       <widget class="QLabel" name="timeLabel">
        <property name="minimumWidth">
         <number>60</number>
        </property>
        <property name="text">
         <string>00:00</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="progressSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="totalTimeLabel">
        <property name="minimumWidth">
         <number>60</number>
        </property>
        <property name="text">
         <string>00:00</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controlsLayout">
      <item>
       <widget class="QPushButton" name="rewindStartButton">
        <property name="text">
         <string>⏮</string>
        </property>
        <property name="toolTip">
         <string>Ir al inicio</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="previousButton">
        <property name="text">
         <string>⏪</string>
        </property>
        <property name="toolTip">
         <string>Anterior</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="playButton">
        <property name="text">
         <string>⏸</string>
        </property>
        <property name="toolTip">
         <string>Reproducir/Pausar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="nextButton">
        <property name="text">
         <string>⏩</string>
        </property>
        <property name="toolTip">
         <string>Siguiente</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="fastForwardButton">
        <property name="text">
         <string>⏭</string>
        </property>
        <property name="toolTip">
         <string>Ir al final</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="volumeLabel">
        <property name="text">
         <string>Volumen</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="volumeSlider">
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="volumePercentLabel">
        <property name="minimumWidth">
         <number>40</number>
        </property>
        <property name="text">
         <string>50%</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="speedComboBox">
        <property name="toolTip">
         <string>Velocidad de reproducción</string>
        </property>
        <item>
         <property name="text">
          <string>0.5x</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>0.75x</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1.0x</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1.25x</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1.5x</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2.0x</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuArchivo">
    <property name="title">
     <string>Archivo</string>
    </property>
    <addaction name="actionAbrirArchivo"/>
    <addaction name="actionAbrirDirectorio"/>
    <addaction name="separator"/>
    <addaction name="actionSalir"/>
   </widget>
   <widget class="QMenu" name="menuTemas">
    <property name="title">
     <string>Temas</string>
    </property>
    <addaction name="actionTemaClaro"/>
    <addaction name="actionTemaOscuro"/>
   </widget>
   <addaction name="menuArchivo"/>
   <addaction name="menuTemas"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbrirArchivo">
   <property name="text">
    <string>Abrir Archivo</string>
   </property>
  </action>
  <action name="actionAbrirDirectorio">
   <property name="text">
    <string>Abrir Directorio</string>
   </property>
  </action>
  <action name="actionSalir">
   <property name="text">
    <string>Salir</string>
   </property>
  </action>
  <action name="actionTemaClaro">
   <property name="text">
    <string>Tema Claro</string>
   </property>
  </action>
  <action name="actionTemaOscuro">
   <property name="text">
    <string>Tema Oscuro</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
